client-js/src/ERC20Client.ts

```diff
@@ -274,4 +277,67 @@ export default class ERC20Client {
       'total_supply'
     ]) as Promise<BigNumber>;
   }
+
+  /**
+   * Waits for the deploy to be confirmed on chain.
+   * Throws `ContractError` if there was operational error, otherwise `Error` if the deploy wasn't successful
+   * @param deploy signed deploy instance or deploy hash
+   * @param timeout
+   * @returns `GetDeployResult`
+   */
+  public async waitForDeploy(
+    deploy: DeployUtil.Deploy | string,
+    timeout?: number
+  ): Promise<GetDeployResult> {
+    const casperClient = new CasperServiceByJsonRPC(this.nodeAddress);
+
+    let result: GetDeployResult;
+
+    // this should be handled in the SDK
+    if (deploy instanceof DeployUtil.Deploy) {
+      result = await casperClient.waitForDeploy(deploy, timeout);
+    } else {
+      const waitForDeploy = async (deployHash: string, t = 60000) => {
+        const sleep = (ms: number) =>
+          // eslint-disable-next-line no-promise-executor-return
+          new Promise(resolve => setTimeout(resolve, ms));
+        const timer = setTimeout(() => {
+          throw new Error('Timeout');
+        }, t);
+        // eslint-disable-next-line no-constant-condition
+        while (true) {
+          // eslint-disable-next-line no-await-in-loop
+          const d = await casperClient.getDeployInfo(deployHash);
+          if (d.execution_results.length > 0) {
+            clearTimeout(timer);
+            return d;
+          }
+          // eslint-disable-next-line no-await-in-loop
+          await sleep(400);
+        }
+      };
+
+      result = await waitForDeploy(deploy);
+    }
+
+    if (
+      result.execution_results.length > 0 &&
+      result.execution_results[0].result.Failure
+    ) {
+      // Parse execution result
+      const { error_message } = result.execution_results[0].result.Failure;
+      const contractErrorMessagePrefix = 'User error: ';
+      if (error_message.startsWith(contractErrorMessagePrefix)) {
+        const errorCode = parseInt(
+          error_message.substring(
+            contractErrorMessagePrefix.length,
+            error_message.length
+          ),
+          10
+        );
+        throw new ContractError(errorCode);
+      } else throw new Error(error_message);
+    }
+    return result;
+  }
```

Comment:
maybe we should have a method like this in `casper-js-sdk` ? I wouldn't add this to contract client :) Especially that it is used only in tests

Reply:
the purpose of this method was to parse contract error, if you don't like it approach then we can create a method like `parseDeployResult`

Reply:
yeah I thought about it, it will be more universal for end users :)